include/igl/random_points_on_mesh.cpp

```diff
@@ -35,6 +34,10 @@ IGL_INLINE void igl::random_points_on_mesh(
   A0.bottomRightCorner(A.size(),1) = A;
   // Even faster would be to use the "Alias Table Method"
   cumsum(A0,1,C);
+  const Scalar Cmax = C(C.size()-1);
+  assert(Cmax > 0 && "Total surface area should be positive");
+  // Why is this more accurate than `C /= C(C.size()-1)` ?
+  for(int i = 0;i<C.size();i++) { C(i) = C(i)/Cmax; }
```

Comment:
I think the reason is: here the last value is explicitly made equal to `Cmax/Cmax` which will more likely be `1`.

Reply:
I still don't get it. I expect it to be _exactly_ 1 in both cases. Why would Eigen's broadcaster not give the same result as this loop?

Reply:
I just tried on the meshes from the tutorial data, and I always get `1` if I replace this line by `C /= C(C.size()-1)`, even for a very large number of samples (10,000,000).